process/dataValidators/txValidator.go

```diff
@@ -116,9 +141,118 @@ func (txv *txValidator) CheckTxValidity(interceptedTx process.TxValidatorHandler
 	return nil
 }
 
+func (txv *txValidator) checkPermission(interceptedTx process.InterceptedTransactionHandler, account state.UserAccountHandler) error {
+	txData, err := getTxData(interceptedTx)
+	if err != nil {
+		return err
+	}
+
+	if account.IsFrozen() {
+		err = txv.checkGuardedTransaction(interceptedTx, account)
+		if err == nil {
+			return nil
+		}
+
+		err = checkOperationAllowedToBypassGuardian(txData)
```

Comment:
If there is no error on the guarded transaction check, it means that the guardian has signed.
This line is reached if the previous check failed, so it will check if the operation is allowed to bypass a missing guardian signature. The only qualifying operation right now is setting a new guardian, which should cover your concern.

Logic should be then, if the set guardian tx was co-signed, then the change can be done immediately( with activation in current epoch) otherwise the activation will be done after the predefined waiting period (x epochs).